integration/spark/build.gradle

```diff
@@ -78,16 +85,12 @@ dependencies {
     implementation "com.fasterxml.jackson.core:jackson-databind:${jacksonVersion}"
     implementation "com.fasterxml.jackson.core:jackson-core:${jacksonVersion}"
     implementation "com.fasterxml.jackson.datatype:jackson-datatype-jsr310:${jacksonVersion}"
-    implementation "com.fasterxml.jackson.module:jackson-module-scala_2.11:${jacksonVersion}"
+    compileOnly "com.fasterxml.jackson.module:jackson-module-scala_2.11:${jacksonVersion}"
```

Comment:
`LogicalPlanSerializer` wouldn't  work with `compileOnly jackson-module-scala_2.11`  for 2.4.7 and for 3.1. Due to `relocate` in `shadowJar`. In compile time we have openlineage prefix, and if we bound it to our jar, it would work fine for 2.4.7. If we run it for spark3, we would get   
```
java.lang.NoSuchMethodError: scala.Product.$init$(Lscala/Product;)V
      at io.openlineage.spark.agent.lifecycle.plan.LogicalRDDVisitorTest.testApply(LogicalRDDVisitorTest.java:49)
```
Cause it loads scala 2.11 compiled classes in 2.12 env

Reply:
Are you saying that `compileOnly` jars are included in the bundled shadow jar? Exclude the jar from the bundle and get rid of the `relocate` statement. The goal is to just use the Jackson dependencies as they are provided by Spark. No renaming,  no bundling.

Reply:
`compileOnly` is not included, but we don't have it in runtime (with prefix), that's why it wouldn't work. Removing `relocate` in in other PR. That's what I meant when say, that this PR relates to [this](https://github.com/OpenLineage/OpenLineage/pull/275).
It should start work, when [PR](https://github.com/OpenLineage/OpenLineage/pull/275) will go live

Reply:
Ok, can you rebase one PR on the other one? We shouldn't know that things will work only after everything is merged. It seems to me the Jackson PR should go first, then the Spark 3 tests can be rebased on top of that one. Then we should be able to validate everything works the way we expect.

Reply:
yes, Jackson PR should go first. I can rebase it when we merge that PR.
Second option I see is to create a separate branch `spark3` and change PR base to it. With that approach we would have a small separate PRs as we discussed previously, and it wouldn't impact `main` and changes in `main` wouldn't impact spark3 support. Later when Jackson PR, this one and Source Tree PRs will be merged, we can pull changes from main, test them and merge to main.

Reply:
I would prefer to have both PR's to start working on Splitting Source Tree Issue. Cause it can cause the same issue as right now.

Reply:
You can simply rebase the `feature/spark3_tests_support` on the `feature/jackson-serialization-test-for-using-provided-libs` branch and change this PR to use the other branch as its base. Then we can see both sets of changes and you'll be able to verify that they both work. We don't have to wait until the other change is merged to do that.

Unfortunately, we're splitting conversations about dependencies and serialization between both of these PRs and it's hard to follow what's going on.

Reply:
ok